server/docs/design/block-verification.md

```diff
@@ -0,0 +1,193 @@
+# Block Verification Design
+
+## Table of Contents
+1. [Purpose](#purpose)
+1. [Goals](#goals)
+1. [Terms](#terms)
+1. [Entities](#entities)
+1. [Design](#design)
+1. [Interfaces](#interfaces)
+1. [Enums](#enums)
+1. [Metrics](#metrics)
+1. [Exceptions](#exceptions)
+
+## Purpose
+The purpose of the Block Verification feature is to ensure that blocks received from consensus nodes are valid and have not been tampered with. This is achieved by re-calculating the block hash and verifying it against the signature provided by the consensus node.
+
+## Goals
+1. The block-node must re-create the block hash from the block items and verify that it matches the hash implied by the signature. 
+1. If verification fails, the block should be considered invalid, and appropriate error-handling procedures must be triggered.
+
+## Terms
+<dl>
+<dt>Consensus Node (CN)</dt><dd>A node that produces and provides blocks.</dd>
+<dt>Block Items</dt><dd>The block data pieces (header, events, transactions, transaction result, state changes, proof) that make up a block.</dd>
+<dt>Block Hash</dt><dd>A cryptographic hash representing the block’s integrity.</dd>
+<dt>Signature</dt><dd>The cryptographic signature of the block hash created by Network private key (aka: LedgerId).</dd>
+<dt>Public Key</dt><dd>The public key of the LedgerId that signed the block.</dd>
+</dl>
+
+## Entities
```

Comment:
I would push back on this one, since Entities have a bullet list and more than 1 line of explanation, not sure on documentation guidelines and best practices, but does not seem right to me to make Entities a definition list.

Reply:
I would say it can go either way.
Whichever approach produces the most readable layout is fine, and I believe we can trust one another (with occasional suggestions for possible alternatives) to choose the best option.

Reply:
@jsync-swirlds +1, @AlfredoG87 up to you.